openml/datasets/dataset.py

```diff
@@ -544,13 +544,13 @@ def _load_data(self):
                     data, categorical, attribute_names = pickle.load(fh)
         except FileNotFoundError:
             raise ValueError(f"Cannot find file for dataset {self.name} at location '{fpath}'.")
-        except (EOFError, ModuleNotFoundError, ValueError) as e:
+        except (EOFError, ModuleNotFoundError, ValueError, AttributeError) as e:
             error_message = e.message if hasattr(e, "message") else e.args[0]
             hint = ""
 
             if isinstance(e, EOFError):
                 readable_error = "Detected a corrupt cache file"
-            elif isinstance(e, ModuleNotFoundError):
+            elif isinstance(e, (ModuleNotFoundError, AttributeError)):
```

Comment:
I'm not sure if they should be in the same `elif` as the hint should point to a different issue.

Reply:
> I don't believe there is an open issue for this.

You could link this PR.

> Do you think it is useful to document each instance separately?

If possible, yes. Will help folks who only skim errors and don't read whole threads.

But in general, we could just have a hint stating that the pandas version most likely doesn't match.

Reply:
Revised.